Framework/LiveData/src/Kafka/KafkaEventStreamDecoder.cpp

```diff
@@ -613,14 +616,15 @@ void KafkaEventStreamDecoder::initLocalCaches(
 
   // Cache spec->index mapping. We assume it is the same across all periods
   m_specToIdx =
-      eventBuffer->getSpectrumToWorkspaceIndexVector(m_specToIdxOffset);
+      eventBuffer->getDetectorIDToWorkspaceIndexVector(m_specToIdxOffset);
```

Comment:
I seem to recall talking to @matthew-d-jones about this before and the mapping should be spectrum number to workspace index.

This is handled or the ESS by creating the dummy 1 to 1 mapping but would break non-ESS use (as demonstrated by the failing tests).